reportportal_client/service.py

```diff
@@ -6,50 +6,75 @@
 
 
 class ReportPortalService(object):
+    """Service class with report portal event callbacks."""
+
     def __init__(self, endpoint, project, token, api_base=None):
+        """Init the service class.
+
+        Args:
+            endpoint: endpoint of report portal service.
+            project: project name to use for launch names.
+            token: authorization token.
+            api_base: defaults to api/v1, can be customized to use another version.
+        """
         super(ReportPortalService, self).__init__()
         self.endpoint = endpoint
         if api_base is None:
             self.api_base = "api/v1"
         self.project = project
         self.token = token
-        self.base_url = os.path.join(self.endpoint,
-                                     self.api_base,
-                                     self.project)
+        self.base_url = self.uri_join_safe(self.endpoint,
+                                           self.api_base,
+                                           self.project)
         self.headers = {"Content-Type": "application/json",
                         "Authorization": "{0} {1}".format("bearer",
                                                           self.token)}
         self.session = requests.Session()
 
+    @staticmethod
+    def uri_join_safe(*uri_parts):
```

Comment:
i would rename this method. 'safe' doesn't means nothing and does not give an idea to reader what this method actually does. In general commit is OK.